beacon-chain/sync/validate_data_column.go

```diff
@@ -23,88 +23,121 @@ import (
 	if err != nil {
 		log.WithError(err).Debug("Ignored column sidecar: could not calculate epoch start slot")
 		return pubsub.ValidationIgnore, nil
 	}
+
 	if startSlot >= ds.SignedBlockHeader.Header.Slot {
 		err := fmt.Errorf("finalized slot %d greater or equal to block slot %d", startSlot, ds.SignedBlockHeader.Header.Slot)
 		log.Debug(err)
 		return pubsub.ValidationIgnore, err
 	}
-	// Handle sidecar when the parent is unknown.
+
+	// [IGNORE] The sidecar's block's parent (defined by block_header.parent_root) has been seen (via both gossip and non-gossip sources) (a client MAY queue sidecars for processing once the parent block is retrieved).
 	if !s.cfg.chain.HasBlock(ctx, [32]byte(ds.SignedBlockHeader.Header.ParentRoot)) {
 		err := errors.Errorf("unknown parent for data column sidecar with slot %d and parent root %#x", ds.SignedBlockHeader.Header.Slot, ds.SignedBlockHeader.Header.ParentRoot)
 		log.WithError(err).Debug("Could not identify parent for data column sidecar")
 		return pubsub.ValidationIgnore, err
 	}
+
+	// [REJECT] The sidecar's block's parent (defined by block_header.parent_root) passes validation.
 	if s.hasBadBlock([32]byte(ds.SignedBlockHeader.Header.ParentRoot)) {
 		bRoot, err := ds.SignedBlockHeader.Header.HashTreeRoot()
 		if err != nil {
 			return pubsub.ValidationIgnore, err
 		}
+
+		// If parent is bad, we set the block as bad.
 		s.setBadBlock(ctx, bRoot)
 		return pubsub.ValidationReject, errors.Errorf("column sidecar with bad parent provided")
 	}
+
+	// [REJECT] The sidecar is from a higher slot than the sidecar's block's parent (defined by block_header.parent_root).
 	parentSlot, err := s.cfg.chain.RecentBlockSlot([32]byte(ds.SignedBlockHeader.Header.ParentRoot))
 	if err != nil {
 		return pubsub.ValidationIgnore, err
 	}
+
 	if ds.SignedBlockHeader.Header.Slot <= parentSlot {
 		return pubsub.ValidationReject, errors.Errorf("invalid column sidecar slot: %d", ds.SignedBlockHeader.Header.Slot)
 	}
+
+	// [REJECT] The current finalized_checkpoint is an ancestor of the sidecar's block -- i.e. get_checkpoint_block(store, block_header.parent_root, store.finalized_checkpoint.epoch) == store.finalized_checkpoint.root.
 	if !s.cfg.chain.InForkchoice([32]byte(ds.SignedBlockHeader.Header.ParentRoot)) {
 		return pubsub.ValidationReject, blockchain.ErrNotDescendantOfFinalized
 	}
 
+	// [REJECT] The sidecar's kzg_commitments field inclusion proof is valid as verified by verify_data_column_sidecar_inclusion_proof(sidecar).
 	if err := blocks.VerifyKZGInclusionProofColumn(ds); err != nil {
 		return pubsub.ValidationReject, err
 	}
 
+	// [REJECT] The sidecar's column data is valid as verified by verify_data_column_sidecar_kzg_proofs(sidecar).
 	verified, err := peerdas.VerifyDataColumnSidecarKZGProofs(ds)
 	if err != nil {
 		return pubsub.ValidationReject, err
 	}
+
 	if !verified {
 		return pubsub.ValidationReject, errors.New("failed to verify kzg proof of column")
 	}
+
+	// ?????
```

Comment:
@nisdas to what specification rule this portion of code does correspond?

Reply:
Actually I was more talking about the following block:

```go
	if err := coreBlocks.VerifyBlockHeaderSignatureUsingCurrentFork(parentState, ds.SignedBlockHeader); err != nil {
		return pubsub.ValidationReject, err
	}
```

Reply:
> [REJECT] The proposer signature of sidecar.signed_block_header, is valid with respect to the block_header.proposer_index pubkey.

Reply:
Fixed in 6431659ef2897c05050697b7112901dce836e965